sdks/python/src/opik/evaluation/metrics/heuristics/rouge.py

```diff
@@ -0,0 +1,129 @@
+from typing import Any, List, Union
+from opik.exceptions import MetricComputationError
+from opik.evaluation.metrics import base_metric, score_result
+
+try:
+    from rouge_score import rouge_scorer
+except ImportError:
+    rouge_scorer = None
+
+
+class ROUGE(base_metric.BaseMetric):
+    """
+    A metric that computes the ROUGE, or Recall-Oriented Understudy for Gisting Evaluation score between an output and reference string mainly used for evaluating text summarization.
+    ROUGE is case insensitive, meaning that upper case letters are treated the same way as lower case letters.
+    This metrics is a wrapper around the Google Research reimplementation of ROUGE, which is based on the `rouge-score` library.
+
+    References:
+        - https://github.com/google-research/google-research/tree/master/rouge
+        - https://huggingface.co/spaces/evaluate-metric/rouge
+
+    Args:
+        name: The name of the metric. Defaults to "rouge_metric".
+        track: Whether to track the metric. Defaults to True.
+        rouge_type: Type of ROUGE score to compute. Defaults to "rouge1". Must be one of the following:
+                    - "rouge1": unigram (1-gram) based scoring
+                    - "rouge2": bigram (2-gram) based scoring
+                    - "rougeL": Longest common subsequence based scoring
+                    - "rougeLSum": splits text using '\\n'"
+        use_stemmer: Whether to use stemming when computing ROUGE. Defaults to False.
+        split_summaries: Whether to split summaries into sentences. Defaults to False.
+        tokenizer: A tokenizer to use when splitting summaries into sentences. Defaults to None.
+
+    Example:
+        >>> from opik.evaluation.metrics import ROUGE
+        >>> rouge_metric = ROUGE()
+        >>> result = rouge_metric.score(
+        ...     output="The quick brown fox jumps over the lazy dog.",
+        ...     reference="The quick brown fox jumps over the lazy dog."
+        ... )
+        >>> print(result.value)
+        1.0
+    """
+
+    def __init__(
+        self,
+        name: str = "rouge_metric",
+        track: bool = True,
+        rouge_type: str = "rouge1",
+        use_stemmer: bool = False,
+        split_summaries: bool = False,
+        tokenizer: Any | None = None,
```

Comment:
`Any | None` -> `Optional[Any]`
`|` typing operator has appeared in python 3.10 only and we support python3.8+

Reply:
Updated the type

Reply:
@alexkuzmik , Is this the reason why some checks were failing?